bdshemu/bdshemu_x86.c

```diff
@@ -3559,6 +3552,13 @@ ShemuX86Emulate(
         }
```

Comment:
The code would be cleared if you would just move the entire `if` immediately after the `switch` statement (this means that other changes are not needed):
```suggestion
        }
        
        // The stop flag has been set, this means we've reached a valid instruction, but that instruction cannot be
        // emulated (for example, SYSCALL, INT, system instructions, etc).
        if (stop)
        {
            return SHEMU_ABORT_CANT_EMULATE;
        }
```

Reply:
I did it now, but then I reverted the changes. I remembered that the reason I didn't do this initially was because FPU instructions that use a memory operand set `stop` to true and do an early `continue`.

```cpp
if (ND_NULL != pMemOp)
{
    ND_UINT64 gla = ShemuX86ComputeLinearAddress(Context, pMemOp);

    if (!ShemuIsShellcodePtr(Context, gla, pMemOp->Size) &&
        !ShemuIsStackPtr(Context, gla, pMemOp->Size))
    {
        stop = ND_TRUE;
    }
}

continue;
```

Reply:
Ah yes, of course, I was looking at my local changes, where the `continue` is instead a `goto` exactly at the end of the `while` loop. 
I'd say it's better to add a `post_emulate` label after the `switch` statement, and then `goto` there from the FPU check, instead of doing a `continue` (I didn't like that `continue` anyway). This way we can just move our `if` around.

Reply:
Done